rfd/0230-component-feature-advertisement.md

```diff
@@ -0,0 +1,306 @@
+experience, even if errors are handled gracefully. This mechanism provides an explicit, composable signal consumers can
+use to "fail silently" when certain capabilities are not available and hide unsupported flows from the get-go.
+
+## Design
+### Payload format
+The payload is simple and forward-compatible by design. In v1, features are Boolean: presence means "supported", absence
+means "not supported". Each feature is identified by an append-only enum value so IDs remain stable over time.
+```protobuf
+enum ComponentFeatureID {
+  COMPONENT_FEATURE_ID_UNSPECIFIED             = 0;
+  // v1 addition: Support for RFD 228 ResourceAccessIDs and Resource Constraints
+  COMPONENT_FEATURE_ID_RESOURCE_CONSTRAINTS_V1 = 1;
+}
+
+message ComponentFeatures {
+  repeated ComponentFeatureID features = 1;
+}
+```
+
+Unknown feature IDs can be ignored by consumers (optionally logged once per service identity). Because senders only
+include "true" values, the payloads stay compact, and omission naturally encodes "false"/"not supported".
+
+### Advertising from services
+Each service that advertises features exposes a list of its current `ComponentFeatures`. For heartbeat-ing services,
+this payload is attached to the presence resources they already send over the inventory control stream (e.g.,
+`AppServerV3` for App Service, `DatabaseServerV3` for DB Service, and so on). No changes are required in the heartbeat
+machinery itself; services simply add a `ComponentFeatures` field to the Spec of the presence resources they are
+already constructing.
+
+A service should advertise a flag only when:
+- the implementation is present in the build, and
+- it is usable with the current configuration and dependencies.
+
+For the initial Resource Constraints work, support is not configuration-dependent. If the code path is compiled into the
+binary, the service can unconditionally advertise the flag for the resources that participate in that flow.
+
+Auth will stores these payloads alongside existing status information so they are available through the same APIs that
+already expose presence and cluster state.
+
+Auth itself won't carry `ComponentFeatures` on an auth-server presence resource. Instead, it can expose its own feature
+support via adding some `AuthComponentFeatures` field to its Spec when heartbeating.
+
+### Consuming feature information
+On the consumer side, the logic is straightforward. For a target operation, the consumer gathers the participating
+services (for example, Auth and the App Service that will serve an app session) and intersects their feature sets to
+determine feature support.
+
+In practice,
+- Consumers obtain per-service `ComponentFeatures` by reading presence resources from Auth (e.g., app servers,
+  database servers, and nodes).
+- Consumers also obtain all present Auth instances (e.g., via `accessPoint.GetAuthServers()` in Web handlers),
+  which contain Auth's own `ComponentFeatures` on their Specs.
```

Comment:
> Consumers also obtain all present Auth instances

Who will the consumers be in practice? Auth servers are currently only cached at Auth and Proxy which would cause this to be a backend read by all other services. We should also not be using GetAuthServers. If a ListAuthServers does not exist we should add one.